### ISSUE TITLE
Icons disappearing?

### ISSUE THREAD

**Issue by jesseleite** [sparksuite/simplemde-markdown-editor#70]
You mention this in your readme:

``` js
// Customize only the order of existing buttons
var simplemde = new SimpleMDE({
    toolbar: ["bold", "italic", "heading", "|", "quote"],
});
```

When I do this, my icons and tooltips disappear.  Is this intentional, or bug?
![Example Screencast](http://g.recordit.co/5U83VhmlpD.gif)

For the record, here's my exact code:

``` js
var simplemde = new SimpleMDE({
    element: document.getElementById("#description"),
    tabSize: 4,
    toolbar: ["bold", "italic", "heading", "|", "quote"],
});
simplemde.render();
```

**Comment by WesCossick** [comment]
Definitely a bug, I'll look into this in more detail and will release a fix soon. Thanks for catching this.

**Comment by jesseleite** [comment]
Awesome thank you man.

**Comment by WesCossick** [comment]
On further testing, I am not able to recreate this issue. I did notice one bug in your code:

``` js
element: document.getElementById("#description"),
```

...should be...

``` js
element: document.getElementById("description"),
```

Apart from that, are you including FontAwesome as a dependency? If so, can you set up a JSFiddle that recreates the issue?

**Comment by jesseleite** [comment]
Hmm thanks for catching that `getElementById("#description")`, sorry about that.

Doesn't make sense to me though now, because I notice that this works:

`element: document.getElementById("###not-an-element"),`

Cleared cache, but somehow SimpleMDE is still able to apply itself to my textarea#description?

**Comment by jesseleite** [comment]
Oh if passed element is found, SimpleMDE is using first textarea on page as default.

> **element:** The DOM element for the textarea to use. Defaults to the first textarea on the page.

I would have thought that if an `element` key within options exists, the default would be overridden, and an error would be thrown if non-existing element is specified.

**Comment by WesCossick** [comment]
Right now, if an element is _not_ found, it just defaults to the first textarea on the page. Seems simpler that way. Anyway, did you find and correct the icons disappearing bug? Or is that still an issue?

**Comment by jesseleite** [comment]
> Right now, if an element is not found, it just defaults to the first textarea on the page.  Seems simpler that way.

@WesCossick, I agree, if no `element` key is explicitly set in options array.  But but if `element` is explicitly set in options, that suggests the user does not want default behaviour.  If the user then makes an error selecting the element (as I did), there's no error in console to reflect erroneous element selector.  Might I suggest a small tweak?
- If no `element` key is explicitly set in options, revert to your default (first textarea in DOM).
- If an explicit `element` key is set in options, override default (but throw error in console if user's selector is erroneous).

> Anyway, did you find and correct the icons disappearing bug?

I'm struggling to reproduce this error today.  I'm not sure what could have changed.  Feel free to close the issue ...sorry about the trouble.

**Comment by iam4x** [comment]
I'm having this, it happens periodically and it always happens for some users.

The class is `undefined` it's hard to reproduce...

**Comment by WesCossick** [comment]
When it happens, the class for each icon is undefined? Can you provide more details about what is happening?

**Comment by jesseleite** [comment]
Yeah I found it weird that I can't reproduce it now.  I swear I didn't change anything in my code that should've affected it... but who knows.

**Comment by ayudh37** [comment]
This issue is occurring for me in my ionic project. 
I tried a simple example but the icons were showing there. But in my ionic project, the icons do not appear. In the inspector, I can see that the class is fa fa-bold. I am setting the autoDownloadFontAwesome to true when I am creating a new simplemde. I am also loading boostrap in my index.html along with the simplemde CDN links of course

**Comment by Egorzaa** [comment]
I have the same issue with icons when I open web page with IE 11. When I open it in Firefox and Chrome it works well. Also earlier it worked well in IE as well.
my webserver shows next:
/libs/fonts/fontawesome-webfont.eot? HTTP/1.1" 404
/libs/fonts/fontawesome-webfont.woff?v=4.7.0 HTTP/1.1" 404
/libs/fonts/fontawesome-webfont.ttf?v=4.7.0 HTTP/1.1" 404
Also I didnt change Simplemde file structure. Only changed the name the root folder to simplemde.
I downloaded Simplemde zip file again and put it in the library but it didn't change anything.

Finally I found the solution: download fonts from: https://cdnjs.com/libraries/font-awesome

**Comment by Egorzaa** [comment]
I have the same issue with icons when I open web page with IE 11. When I open it in Firefox and Chrome it works well. Also earlier it worked well in IE as well.
my webserver shows next:
/libs/fonts/fontawesome-webfont.eot? HTTP/1.1" 404
/libs/fonts/fontawesome-webfont.woff?v=4.7.0 HTTP/1.1" 404
/libs/fonts/fontawesome-webfont.ttf?v=4.7.0 HTTP/1.1" 404
Also I didnt change Simplemde file structure. Only changed the name the root folder to simplemde.
I downloaded Simplemde zip file again and put it in the library but it didn't change anything.

Finally I found the solution: download fonts from: https://cdnjs.com/libraries/font-awesome

**Comment by wenyAlab** [comment]
I meet this issue  in my project, and I find these icons missing their  styles, thanks for @Egorzaa !  Link [font-awesome ]( https://cdnjs.com/libraries/font-awesome)styles
in your index.html, just like this:
```
<head>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/simplemde/latest/simplemde.min.css">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.css">
</head>

```

**Comment by suhailkc** [comment]
I have solved the issue.

I had the same issue. and I realised that the icons don't work on offline (no internet).
I had experienced this icom hidden issue when my internet temporary blocked. after internet connected the icons came back.

This plugin is using font-awesome cdn for icons.
if your icons are missing, download and use it or use 
`<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.css">`
